tof/models.py

```diff
@@ -133,10 +133,8 @@ def add_translation_to_class(self, trans_mng=None):
                 trans_mng.contribute_to_class(cls, 'objects')
                 origin.contribute_to_class(cls, 'objects_origin')
 
-        cls._meta._field_tof = {
-            'by_id': {self.id: self},
-            'by_name': {self.name: self}
-        }
+        cls._meta._field_tof['by_id'].update({self.id: self})
```

Comment:
@mom1 This should do the work, originally there was `setattr` with nested `setdefault`. It was quite hard to read and understand.  Also, I hope that a good idea to add more tests for core functionality.